### PR TITLE
Print Kani version

### LINKED ISSUE #2570: Output - Kani version
Requested feature: Print the version of Kani at the beginning of a run.
Use case: Kani doesn't print its version despite it printing other versions (for example, CBMC's version gets printed once for each harness). This will provide a better experience for both users and developers.
Link to relevant documentation (Rust reference, Nomicon, RFC): N/A

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

Dusting this off since it'd be great to have for next release.

This PR changes Kani so it prints its version and the invocation type/mode as
`Kani Rust Verifier <version> (<invocation>)` where `<invocation>` is either `standalone` or `cargo plugin`.

Please note that I'm leaving development version information for later in #2617. We can continue the discussion we started in #2571 here.

### Resolved issues:

Resolves #2570 

### Testing:

* How is this change tested? Two new tests.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-driver/build.rs [hunk 1]
```diff
@@ -3,11 +3,28 @@
...
+    // Force this script to re-run if `kani-verifier/Cargo.toml` changed.
+    // Otherwise, we may not get the up-to-date version in `kani-driver`.
+    println!("cargo:rerun-if-changed=../Cargo.toml");
+
+    // Pass the `kani-verifier` version through the build environment.
+    println!("cargo:rustc-env=KANI_VERIFIER_VERSION={}", kani_verifier_version());
+}
+
+/// Retrieve the `kani-verifier` version using `cargo metadata`
+fn kani_verifier_version() -> String {
```
Comment (celinval) [inline kani-driver/build.rs:25]: Shouldn't the `kani-driver` version always match `kani-verifier`? I don't understand why we are doing this.

File: kani-driver/src/main.rs [hunk 2]
```diff
@@ -77,6 +82,11 @@ fn cargokani_main(input_args: Vec<OsString>) -> Result<()> {
         return assess::run_assess(session, assess::AssessArgs::default());
     }
 
+    if !session.args.common_args.quiet {
+        let kani_version = print_kani_version();
```
Comment (celinval) [inline kani-driver/src/main.rs:86]: Shouldn't this method print the version? It has `print` in the name. :)

File: kani-driver/src/version.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::{determine_invocation_type, InvocationType};
+
+const KANI_RUST_VERIFIER: &str = "Kani Rust Verifier";
+const KANI_VERIFIER_VERSION: &str = env!("KANI_VERIFIER_VERSION");
+
+/// Print Kani version. At present, this is only release version information.
```
Comment (celinval) [inline kani-driver/src/version.rs:9]: This doesn't print though

File: kani-driver/src/version.rs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+/// Print Kani release version as `Kani Rust Verifier <version> (<invocation>)`
+/// where:
+///  - `<version>` is the `kani-verifier` version
+///  - `<invocation>` is `cargo plugin` if Kani was invoked with `cargo kani` or
+///    `standalone` if it was invoked with `kani`.
+fn kani_version_release() -> String {
+    let mut version_str = "Kani Rust Verifier ".to_string();
+    version_str.push_str(KANI_VERIFIER_VERSION);
+
+    let invocation_str = match determine_invocation_type(Vec::from_iter(std::env::args_os())) {
```
Comment (celinval) [inline kani-driver/src/version.rs:25]: nit: you could get this as an argument.

**Review by celinval — COMMENTED** [review]
Shouldn't we print the same blob for all sub-commands as well?

File: kani-driver/src/main.rs (hunk 2 above)
Comment (adpaco-aws) [reply kani-driver/src/main.rs]: True, it prints now.

File: kani-driver/src/version.rs (hunk 4 above)
Comment (adpaco-aws) [reply kani-driver/src/version.rs]: Done.

**Comment by adpaco-aws** [comment]
Addressed all comments.

> Shouldn't we print the same blob for all sub-commands as well?

To be honest, this wasn't clear to me. Guessing it's because I'd really like `assess` to be a tool rather than a sub-command...

File: kani-driver/build.rs (hunk 1 above)
Comment (adpaco-aws) [reply kani-driver/build.rs]: I've opened #2626 to ensure these are the same versions. We just use the CARGO_PKG_VERSION now.

**Review by celinval — APPROVED** [review]
Thanks

File: kani-driver/src/version.rs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    println!("{kani_version}");
+}
+
+/// Print Kani release version as `Kani Rust Verifier <version> (<invocation>)`
+/// where:
+///  - `<version>` is the `kani-verifier` version
+///  - `<invocation>` is `cargo plugin` if Kani was invoked with `cargo kani` or
+///    `standalone` if it was invoked with `kani`.
+fn kani_version_release(invocation_type: InvocationType) -> String {
+    let mut version_str = "Kani Rust Verifier ".to_string();
```
Comment (celinval) [inline kani-driver/src/version.rs:26]: do you ever use this?

Comment (adpaco-aws) [reply kani-driver/src/version.rs]: Good catch. I'm surprised the compiler/`clippy` didn't complain about this.

**Pull request merged**